tests/models/test_boundary_condition.py

```diff
@@ -102,3 +108,205 @@ def test_boundary_condition_mixin(t_end: int):
         # Projecting the expected value to the subdomain.
         expected_val = bg.projection().T @ expected_val
         assert np.allclose(bc_val_prev_ts[bc_type.is_neu], expected_val[bc_type.is_neu])
+
+
+"""Here follows mixins related to testing of Robin limit cases, and eventually the test itself. """
+
+
+class BCValues:
+    """Set boundary values for momentum, mass, and mass and energy balance."""
+
+    def bc_values_displacement(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Assigns displacement values in the x-direction of the west boundary."""
+        values = np.ones((self.nd, bg.num_cells))
+        bounds = self.domain_boundary_sides(bg)
+        values[0][bounds.west] += np.ones(len(values[0][bounds.west]))
+        return values.ravel("F")
+
+    def bc_values_pressure(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Assigns pressure values on the west boundary."""
+        values = np.ones(bg.num_cells)
+        bounds = self.domain_boundary_sides(bg)
+        values[bounds.west] += np.ones(len(values[bounds.west]))
+        return values
+
+    def bc_values_temperature(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Assigns temperature values on the west boundary."""
+        values = np.ones(bg.num_cells)
+        bounds = self.domain_boundary_sides(bg)
+        values[bounds.west] += np.ones(len(values[bounds.west]))
+        return values
+
+
+class BCRobDir:
+    """Set Dirichlet and Robin for momentum, mass, and mass and energy balance.
+
+    Sets Dirichlet on the west boundary, and Robin with alpha = 0 on all other
+    boundaries.
+
+    """
+
+    def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:
+        bounds = self.domain_boundary_sides(sd)
+        bc = pp.BoundaryConditionVectorial(
+            sd,
+            bounds.north + bounds.south + bounds.east + bounds.west,
+            "rob",
+        )
+        bc.is_rob[:, bounds.west] = False
+        bc.is_dir[:, bounds.west] = True
+
+        r_w = np.tile(np.eye(sd.dim), (1, sd.num_faces))
+        bc.robin_weight = np.reshape(r_w, (sd.dim, sd.dim, sd.num_faces), "F") * 0
+        return bc
+
+    def bc_type_scalar_rob_dir(self, sd: pp.Grid) -> pp.BoundaryCondition:
```

Comment:
If you rename this method to e.g. `bc_type_scalar`, you can let the below class inherit from this, avoiding redefinition of `bc_type_x_flux` methods.

Reply:
Maybe I misunderstand something here, but I do not see how that will help the case. The bc_type_scalar methods are different (one assigns Neumann, one assigns Robin), so redefinition is needed in some sense anyways? Or?